lib/user.rb

```diff
@@ -0,0 +1,29 @@
+require 'bcrypt'
+class User
+
+  include DataMapper::Resource
+
+  attr_reader :password
+  attr_accessor :password_confirmation
+
+  validates_confirmation_of :password
+
+  property :id, Serial
+  property :email, String,
+           unique: true,
+           message: 'This email address has be taken'
+  property :password_digest, Text
+  property :name, String, unique: true, message: 'This name has be taken'
+  property :username, String
+
+  def password=(password)
+    @password = password
+    self.password_digest = BCrypt::Password.create(password)
+  end
+
+  def self.authenticate(email, password)
+    user = first(email: email)
+    user if user && BCrypt::Password.new(user.password_digest) == password
+  end
+
+end
```

Comment:
And on the page would be nice to see something like this:

'Hello!"' peep created by 'Rich'
       "Hello, Richard! I'm Mihai!" reply created by 'Mihai'
       "How are you?"                   reply created by 'Rich'
"It's such a beautiful day today" peep created by 'Mihai'
       "We had better"                  reply created by 'Rich'

........